R/tm_data_table.R

```diff
@@ -325,22 +333,29 @@ ui_data_table <- function(id,
 }
 
 srv_data_table <- function(id,
-                           datasets,
+                           data,
+                           filter_panel_api,
                            dataname,
                            if_filtered,
                            if_distinct,
                            dt_args,
                            dt_options) {
+
   moduleServer(id, function(input, output, session) {
     output$data_table <- DT::renderDataTable(server = FALSE, {
       variables <- input$variables
 
       validate(need(variables, "need valid variable names"))
 
-      df <- datasets$get_data(
-        dataname,
-        filtered = if_filtered()
-      )
+      observeEvent(if_filtered(), {
```

Comment:
as discussed, we should move the 
"use/don't use filtering" out of tmg and into teal.slice